File: 1939-queries-on-number-of-points-inside-a-circle/1939-queries-on-number-of-points-inside-a-circle.cpp
class Solution {
public:
    int funcCircle(vector<int> point,vector<int> query){
        double S =(point[0] - query[0]) * (point[0] - query[0]) +
               (point[1] - query[1]) * (point[1] - query[1]) - query[2] * query[2];
        if(S<=0){
            return 1;
        }
        else{
            return 0;
        }
    }
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        vector<int>result;
        int count=0;
        for(int q=0; q<queries.size();q++){
            for(int p=0;p<points.size();p++){
                count+=funcCircle(points[p], queries[q]);
            }
            result.push_back(count);
            count=0;
        }
        return result;

        
    }
};

File: 1939-queries-on-number-of-points-inside-a-circle/1939-queries-on-number-of-points-inside-a-circle.java
class Solution {
    public int[] countPoints(int[][] points, int[][] queries) {
        int n = queries.length;
        int[] res = new int[n];

        for (int i = 0; i < n; i++)
            res[i] = getAns(points, queries, i);
        return res;
    }

    public int getAns(int[][] points, int[][] queries, int i){
            int count = 0, centerX = queries[i][0], centerY = queries[i][1], radius = queries[i][2];

            for (int j = 0; j < points.length; j++) {
                int pointX = points[j][0], pointY = points[j][1];
                if ((pointX - centerX) * (pointX - centerX) + (pointY - centerY) * (pointY - centerY) <= radius * radius) count++;
            }
            return count;
    }
}